src/workerd/api/streams/internal.c++

```diff
@@ -2333,14 +2363,24 @@ void IdentityTransformStreamImpl::abort(kj::Exception reason) {
       request.fulfiller->reject(kj::cp(reason));
     }
     KJ_CASE_ONEOF(request, WriteRequest) {
-      KJ_FAIL_ASSERT("abort() is supposed to wait for any pending write() to finish");
+      switch (option) {
+        case AbortOption::NONE: {
+          KJ_FAIL_ASSERT("abort() is supposed to wait for any pending write() to finish");
+        }
+        case AbortOption::ABORT_QUEUED_WRITES: {
+          request.fulfiller->reject(kj::cp(reason));
```

Comment:
Now that all the write requests are canceled first, is this needed?

Maybe we no longer need to pass `AbortOption` down to the `WritableStreamSink`?